elastic/kibana
# [SECURITY] add telemetry on authentication type

## Summary

Resolves: https://github.com/elastic/kibana/issues/117517

## To Verify
1. Run this branch and login with different authentication type
2. Go to https://localhost:5601/api/stats?extended=true&legacy=true and verify that usage counters have been updated with the expected numbers. Usage counter section will look something like this:

```
{
  "usage": {
    "usage_counters": {
      "dailyEvents": [
        {
          "domainId": "security",
          "counterName": "realm",
          "counterType": "SecurityAuthType",
          "lastUpdatedAt": "2021-08-16T02:01:16.786Z",
          "fromTimestamp": "2021-08-16T00:00:00Z",
          "total": 1
        },
        {
          "domainId": "security",
          "counterName": "token",
          "counterType": "SecurityAuthType",
          "lastUpdatedAt": "2021-08-16T02:01:56.790Z",
          "fromTimestamp": "2021-08-16T00:00:00Z",
          "total": 1
        }
      ]
    }
  }
}
```


### Checklist
